libs/dsb-address-book-nats-js/src/nats-jetstream-address-book.ts

```diff
@@ -0,0 +1,51 @@
+import { ChannelMetadata, IAddressBook } from '@energyweb/dsb-address-book-core';
+import { ITransport } from '@energyweb/dsb-transport-core';
+import { IAM } from 'iam-client-lib';
+
+export { IAddressBook, ChannelMetadata };
+export class NatsJetstreamAddressBook implements IAddressBook {
+    private readonly addressBookChannel = 'channels.dsb.apps.energyweb.iam.ewc';
+    private readonly cache = new Map<string, ChannelMetadata>();
+    private iam: IAM;
+
+    constructor(
+        private readonly transport: ITransport,
+        private readonly web3Url: string,
+        private readonly privateKey: string,
+        private readonly mbDID: string
+    ) {
+        this.iam = new IAM({ rpcUrl: this.web3Url, privateKey: this.privateKey });
+    }
+
+    public async init() {
+        await this.iam.initializeConnection({ initCacheServer: false });
+
+        const abChannelIsAvailable = await this.transport.hasChannel(this.addressBookChannel);
+        if (!abChannelIsAvailable) await this.transport.createChannel(this.addressBookChannel);
+
+        const adConsumerIsAvaiable = await this.transport.hasConsumer(
+            this.addressBookChannel,
+            this.mbDID
+        );
+        if (adConsumerIsAvaiable)
+            await this.transport.removeConsumer(this.addressBookChannel, this.mbDID);
+
+        this.transport.subscribe(this.addressBookChannel, this.mbDID, async (msg: any) => {
+            const claim: any = await this.iam.decodeJWTToken({ token: msg.data });
+            this.cache.set(claim.sub, claim.claimData);
+        });
+    }
+
+    public async findByFqcn(fqcn: string): Promise<ChannelMetadata> {
+        return this.cache.get(fqcn);
+    }
+
+    public async register(fqcn: string, channelMetadata: ChannelMetadata): Promise<void> {
+        const claim = await this.iam.createPublicClaim({
+            subject: fqcn,
```

Comment:
just trying to understand, the addressBookChannel being `channels.dsb.apps.energyweb.iam.ewc` on which we're publishing new fcqn, couple of questions:
- can the fqcn be anything or must it be suffixed with `.{addressBookChannel}`?
- when I spin up a new message broker and read/consume from the address book, that's the only time I'll be able to read those channels? If my broker restarts I will have lost the channels?

Reply:
there is no change in fqcn. it is like before.
when mb starts, first checks for old consumer with its DID and tries to delete it first. in this way it will receive all message anytime